### PR TITLE
[Docs] Add RL example for RDT, vLLM, FSDP2, and GRPO.

### PR DESCRIPTION
**Pull request by crypdick**
## Description
Demonstrates RDT in using RL post-training of an LLM against the GSM8K dataset.

## Related issues
Companion PR to https://github.com/ray-project/ray/pull/57961

### CODE REVIEW

File: doc/source/ray-core/api/rdt-rl-vllm-grpo-fsdp2/main.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+from data_util import load_gsm8k_dataset, Scorer
+
+
+def _batch_iter(items: list[tuple[str, str]], batch_size: int):
+    for i in range(0, len(items), batch_size):
+        yield items[i : i + batch_size]
+
+
+@ray.remote(name=REGISTRY_NAME)
+class RayObjectRefRegistry:
```
Comment (crypdick) [inline doc/source/ray-core/api/rdt-rl-vllm-grpo-fsdp2/main.py:39]: This is my workaround for sending the RDT ObjectRef to the vLLM workers. `collective_rpc()` is unable to send the ObjectRef because of the way it serializes objects: https://gist.github.com/crypdick/8bd703085f5c8f8b2f4d2def58bac516

File: doc/source/ray-core/api/rdt-rl-vllm-grpo-fsdp2/learner.py [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+
+        clipped_norm = nn.utils.clip_grad_norm_(self.model.parameters(), GRAD_CLIP_NORM)
+
+        # Skip optimizer step if gradients are effectively zero to avoid NaN weights.
+        # This can happen when all advantages are zero. This can happen if none of the answers are correct.
+        if clipped_norm > 1e-8:
+            self.optim.step()
+        else:
+            print(
+                "[WARNING] Skipping optimizer step due to zero gradients - all samples likely have same reward",
```
Comment (crypdick) [inline doc/source/ray-core/api/rdt-rl-vllm-grpo-fsdp2/learner.py:291]: Without this, after the first training step the Generator state collapses and outputs nothing but end of sentence tokens: `<s><s><s><s><s><s><s>`. One simple way to fix this is to give smoother rewards, i.e. instead of giving all wrong answers 0 reward, giving a bit more reward if the answer is closer to the target.

File: doc/source/ray-core/api/rdt-rl-vllm-grpo-fsdp2/learner.py [hunk 3]
```diff
@@ -0,0 +1,404 @@
...
+            # Non-FSDP fallback.
+            state_dict = self.model.state_dict()
+
+        # Send the state dict to the Ray object store via NIXL in chunks.
+        registry_handle = ray.get_actor(REGISTRY_NAME, namespace=RAY_NAMESPACE)
+        ray.get(registry_handle.reset.remote())
+        for name, weight in state_dict.items():
+            # FIXME: Qiaolin, remove deepcopy once the bug with sending the same weights multiple times is fixed.
+            name_weight = (name, weight)
+            name_weight = copy.deepcopy(name_weight)
```
Comment (Qiaolin-Yu) [inline doc/source/ray-core/api/rdt-rl-vllm-grpo-fsdp2/learner.py:355]: Do we still need to use deepcopy here?

Comment (crypdick) [reply doc/source/ray-core/api/rdt-rl-vllm-grpo-fsdp2/learner.py:355]: @Qiaolin-Yu It was necessary at the time I opened the PR, I am not sure if it has been fixed since then
